src/foam/nanos/fs/fileDropZone/FileCard.js

```diff
@@ -141,22 +152,23 @@ foam.CLASS({
       code: function(X) {
         X.removeFile(X.fileCard.index);
       }
-    }
-  ],
-
-  listeners: [
+    },
     {
       name: 'viewFile',
       code: function() {
-        var blob = this.data.data;
+        debugger;
+        var blob = this.data;
```

Comment:
We pass file as data to fileCard. Why can we access it with just this?

Reply:
Not sure but this seems to work while using `this.data.data` does not

Reply:
Nvm, figured it out. since it is now an action it is rendered in the context of the file card so this.dat for the view is actually this where the action is rendered